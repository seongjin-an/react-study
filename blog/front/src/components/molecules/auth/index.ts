export type {AuthFormProps} from './AuthForm'
export {AuthForm} from './AuthForm'
export type {AuthFooterProps} from './AuthFooter'
export {AuthFooter} from './AuthFooter'