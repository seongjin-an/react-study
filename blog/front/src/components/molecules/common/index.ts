export {PostSubInfo} from './PostSubInfo'
export {Tags} from './Tags'