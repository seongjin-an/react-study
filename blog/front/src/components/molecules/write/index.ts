export {QuillWrapper} from './QuillWrapper'
export {TagList} from './TagList'
export {TagItem} from './TagItem'
export {TagForm} from './TagForm'