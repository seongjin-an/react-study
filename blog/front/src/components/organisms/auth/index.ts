export type {AuthAreaProps} from './AuthArea'
export {AuthArea} from './AuthArea'