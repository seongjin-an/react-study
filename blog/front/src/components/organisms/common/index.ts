export {Responsive} from './Responsive'
export {Header} from './Header'
export {AskModal} from './AskModal'