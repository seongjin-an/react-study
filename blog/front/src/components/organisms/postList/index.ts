export {PostListArea} from './PostListArea'
export {Pagination} from './Pagination'