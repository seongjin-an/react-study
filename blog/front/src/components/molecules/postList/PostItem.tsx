import React from "react";
import styled from "styled-components";
import {Link} from 'react-router-dom'
import palette from "../../../libs/styles/palette";
import {PostSubInfo, Tags} from "../common";
import {IPost} from "../../../modules/posts/postsReducer";
interface IPostItemProps{
    post: IPost
}
export const PostItem = ({post}: IPostItemProps) => {
    const { publishedDate, user, tags, title, body, _id } = post
    return(
        <StyledPostItemBlock>
            <h2>
                <Link to={`/@${user?.username}/${_id}`}>{title}</Link>
            </h2>
            <PostSubInfo user={user?.username} publishedDate={publishedDate} hasMarginTop={false}/>
            <Tags tags={tags}/>
            <p>{body}</p>
        </StyledPostItemBlock>
    )
}
const StyledPostItemBlock = styled.div`
  padding-top: 3rem;
  padding-bottom: 3rem;
  &:first-child{
    padding-top: 0;
  }
  & + &{
    border-top: 1px solid ${palette.gray[2]};
  }
  h2{
    font-size: 2rem;
    margin-bottom: 0;
    margin-top: 0;
    &:hover{
      color: ${palette.gray[6]};
    }
  }
  p{
    margin-top: 2rem;
  }
`