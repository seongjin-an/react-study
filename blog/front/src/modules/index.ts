import {combineReducers, Reducer} from "redux";
import {configureStore} from "@reduxjs/toolkit";
import createSagaMiddleware from "@redux-saga/core";
import {all, fork} from "@redux-saga/core/effects";
import auth from "./auth/authReducer";
import loading from './loading/loadingReducer'
import user from './user/userReducer'
import write from './write/writeReducer'
import post from "./post/postReducer";

import authSaga from "./auth/authSaga";
import userSaga from "./user/userSaga";
import writeSaga from "./write/writeSaga";
import postSaga from "./post/postSaga";

const rootReducer:Reducer = combineReducers({
    auth,
    loading,
    user,
    write,
    post
})
export const rootSaga = function*(){
    yield all([
        fork(authSaga),
        fork(userSaga),
        fork(writeSaga),
        fork(postSaga)
    ])
}
export type RootState = ReturnType<typeof rootReducer>
export const createStore = () => {
    const sagaMiddleware = createSagaMiddleware();
    const store = configureStore({
        reducer: rootReducer,
        middleware: [sagaMiddleware],
        devTools: process.env.NODE_ENV !== "production",
    })
    sagaMiddleware.run(rootSaga);
    return store
}

export default rootReducer