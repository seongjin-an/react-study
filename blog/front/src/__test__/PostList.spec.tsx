import React from "react";
// import {applyMiddleware, createStore} from 'redux'
import {createStore} from "../modules";
import {configureStore, Store} from "@reduxjs/toolkit";
import {configure, mount} from "enzyme";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import {PostListPage} from "../pages/postList";
import toJson from "enzyme-to-json";
import "@testing-library/jest-dom/extend-expect";
import {PostListArea} from "../components/organisms/postList";
import {PostItem} from "../components/molecules/postList";
import rootReducer, {rootSaga} from "../modules";
import createSagaMiddleware from "@redux-saga/core";

let store:Store
const setup = (WrappedComponent: React.ComponentProps<any>, props={}) => {
    const wrapper = mount(
        <Provider store={store}>
            <BrowserRouter>
                <WrappedComponent {...props} />
            </BrowserRouter>
        </Provider>
    )
    return {
        wrapper
    }
}
describe('post list component', () => {
    beforeEach(async () => {
        // store = createStore()
        const sagaMiddleware = createSagaMiddleware();
        const _store = configureStore({
            reducer: rootReducer,
            middleware: [sagaMiddleware],
            devTools: process.env.NODE_ENV !== "production",
        })
        sagaMiddleware.run(rootSaga);
        store = _store
        configure({ adapter: new Adapter() });
    })
    it('should match snapshot', async () => {
        const props = {
            location:{
                search: ''
            },
            match:{
                params: {
                    username: ''
                }
            }
        }
        const { wrapper } = await setup(PostListPage)
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    it('should render', async () => {
        const props = {
            location:{
                search: ''
            },
            match:{
                params: {
                    username: ''
                }
            }
        }
        const { wrapper } = await setup(PostListPage, props);
        const header=wrapper.find('.post-list-header');
        expect(header.text()).toBe('REACTERS')
        const writeButton = wrapper.find("CommonButton")
        expect(writeButton.text()).toBe('로그인')
    })
    it('should render post list area', async () => {
        const props = {
            location:{
                search: ''
            },
            match:{
                params: {
                    username: ''
                }
            }
        }

        const { wrapper } = await setup(PostListArea, props)
        const postItems = wrapper.find('div[className="real-post-list"]')
        console.log('real-post-list:', postItems)
        // expect(postItems).toHaveLength(3)
        // expect(postItems.length).toEqual(3)
    })
})
