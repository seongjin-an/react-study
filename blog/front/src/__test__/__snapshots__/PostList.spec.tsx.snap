// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post list component should match snapshot 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <PostListPage>
        <PostListTemplate>
          <Header>
            <styled.div>
              <div
                className="sc-eCstlR eosEnN"
              >
                <Styled(Responsive)>
                  <Responsive
                    className="sc-jSgvzq gnxYRS"
                  >
                    <styled.div
                      className="sc-jSgvzq gnxYRS"
                    >
                      <div
                        className="sc-bdfBQB cPQyqM sc-jSgvzq gnxYRS"
                      >
                        <Link
                          className="logo"
                          to="/"
                        >
                          <LinkAnchor
                            className="logo"
                            href="/"
                            navigate={[Function]}
                          >
                            <a
                              className="logo"
                              href="/"
                              onClick={[Function]}
                            >
                              <span
                                className="post-list-header"
                              >
                                REACTERS
                              </span>
                            </a>
                          </LinkAnchor>
                        </Link>
                        <div
                          className="right"
                        >
                          <CommonButton
                            to="/login"
                          >
                            <Styled(Link)
                              to="/login"
                            >
                              <Link
                                className="sc-dlfnuX gwwbGh"
                                to="/login"
                              >
                                <LinkAnchor
                                  className="sc-dlfnuX gwwbGh"
                                  href="/login"
                                  navigate={[Function]}
                                >
                                  <a
                                    className="sc-dlfnuX gwwbGh"
                                    href="/login"
                                    onClick={[Function]}
                                  >
                                    로그인
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </Styled(Link)>
                          </CommonButton>
                        </div>
                      </div>
                    </styled.div>
                  </Responsive>
                </Styled(Responsive)>
              </div>
            </styled.div>
          </Header>
          <withRouter(PostListArea)>
            <PostListArea
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            >
              <Styled(Responsive)>
                <Responsive
                  className="sc-dQoVA eLMyWI"
                >
                  <styled.div
                    className="sc-dQoVA eLMyWI"
                  >
                    <div
                      className="sc-bdfBQB cPQyqM sc-dQoVA eLMyWI"
                    >
                      <styled.div>
                        <div
                          className="sc-bqyKOL fNwuQZ"
                        />
                      </styled.div>
                    </div>
                  </styled.div>
                </Responsive>
              </Styled(Responsive)>
            </PostListArea>
          </withRouter(PostListArea)>
          <withRouter(Pagination)>
            <Pagination
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
            />
          </withRouter(Pagination)>
        </PostListTemplate>
      </PostListPage>
    </Router>
  </BrowserRouter>
</Provider>
`;
